compiler/passes/cleanup.cpp

```diff
@@ -206,6 +206,20 @@ static void add_parens_to_deinit_fns(FnSymbol* fn) {
 }
 
 
+/* Change the type of the receiver in type methods from "_any" to
+   the type that defined the method.
+*/
+static void set_receiver_type_in_type_methods(FnSymbol* fn) {
+  if (fn->hasFlag(FLAG_METHOD) && fn->thisTag == INTENT_TYPE) {
+    if (TypeSymbol* ts = toTypeSymbol(fn->defPoint->parentSymbol)) {
+      if (fn->_this->type == dtAny) {
+        fn->_this->type = ts->type;
+      }
```

Comment:
As Mike and Michael will attest, I've been on a tear about secondary methods recently.  If it doesn't work, feel free to just add a future/issue for the secondary method case and move on.